### ISSUE TITLE
Support for configuring default and maximum command timeouts

### ISSUE THREAD

**Issue by dgollahon** [buildkite/terraform-provider-buildkite#463]
I would ideally like to configure these settings via terraform:
![image](https://github.com/buildkite/terraform-provider-buildkite/assets/4211073/c030087e-bf68-42a9-bfbb-c806cbcf8946)

I know there is not currently an API available for this either but it would be nice to have the API -> terraform capability.

Thanks!

**Comment by james2791** [comment]
Hey @dgollahon!

Tanks for the request - and hope you had a great New Year to welcome in 2024!

As of now, the provider doesn't have that functionality in place as you've seen as the relevant APIs dont exist _yet_ for setting the upper limit timeouts (UI only at present). Shall will keep you in the loop on if/when that can be added for the provider to also maintain in a stateful manner.

**Comment by mcncl** [comment]
I've raised this internally to be supported in the API(s), we can't move forward with this without API support. Once that support is complete then we can card this up.
